### ISSUE TITLE
Start from Basic Terminology

### ISSUE THREAD

**Issue by mithra-chips** [mithra-chips/ML_blog#1]
<img width="610" alt="Image" src="https://github.com/user-attachments/assets/6328f27c-a96b-49a4-b6b1-8c1f6abc7e4d" />

🔹 Basic Terminology from Chapter 1
This section introduces basic terminology from Chapter 1, along with several illustrative examples to aid understanding.

  1. **Hypothesis Space**: Suppose a **training set** as follows:
![Image](https://github.com/user-attachments/assets/67d94a96-60fd-4164-9cd9-ae8569ecded5)

**Hypothesis Space** contains all combinations of attributes.

![Image](https://github.com/user-attachments/assets/0ce90201-9442-4654-871e-294eb352894f)

The machine learning algorithms try to give out a model from a **version space**.

![Image](https://github.com/user-attachments/assets/866ef31d-d764-4bbc-aa97-b5a3240f4404)

  2. let D={X1,X2,X3...Xm}, meaning m **instances** in data set D.
  3. for each Xi={Xi1, Xi2, Xi3,..,Xid}, meaning d **attributes** exist. Each Xi is a **feature vector**, therefore D is a **feature space**.

- The process for machine learning is as follows:
  1. choose a model for training;
  2. train the model by **training set**;
  3. evaluate the model for validation by **test set**;
  4. predict results for actual data set.
